Package Request: Oilshell

http://www.oilshell.org/
Releases are at https://www.oilshell.org/releases.html
I tried to make my own template file, but I don't know enough about xbps-src to get it working. Here's what I have, maybe it can speed things up a bit
```shell
# Template file for 'oilshell'
pkgname=oilshell
version=0.6.0
revision=2
wrksrc=oil-${version}
#archs="i686 x86_64"
build_style=gnu-configure
hostmakedepends="gcc"
makedepends="bash make readline"
#depends=""
short_desc="Oil, a new unix shell."
maintainer=" <>"
license="Apache"
homepage="http://www.oilshell.org/"
distfiles="https://${pkgname}.org/download/oil-${version}.tar.xz"
checksum=141e3a5c1997a696d65747966978aaa38921d77e303aad9e77b4ab4aedab84b7

do_configure() {
	./configure --prefix=/usr
}

do_build() {
	make
}

do_install() {
	vbin _bin/oil.ovm
}
```

And I wrote a working Arch PKGBUILD which might also be useful:
```shell
pkgname=oilshell
pkgver=0.6.0
pkgrel=3
license=('Apache')
pkgdesc="Oil, a new unix shell"
makedepends=(
	'wychmire-devel'
	'readline')
arch=('i686' 'x86_64')
url="https://www.oilshell.org/"
source=("https://www.oilshell.org/download/oil-$pkgver.tar.xz")
install=oilshell.install
sha256sums=('141e3a5c1997a696d65747966978aaa38921d77e303aad9e77b4ab4aedab84b7')
provides=('oilshell')
conflicts=('oilshell')

build() {
	cd "oil-$pkgver"
	./configure --prefix=/usr --with-readline
	make
}

package() {
	cd "oil-$pkgver"
	# the install script doesn't support DESTDIR
	install -Dm755 _bin/oil.ovm "$pkgdir/usr/bin/oil.ovm"
	ln -s /usr/bin/oil.ovm "$pkgdir/usr/bin/osh"
	ln -s /usr/bin/oil.ovm "$pkgdir/usr/bin/oshc"
	# for some reason osh expects itself to be in /usr/local/bin
	mkdir -p "$pkgdir/usr/local/bin"
	ln -s "$pkgdir/usr/bin/oil.ovm" "$pkgdir/usr/local/bin/osh"
}
```
(oilshell.install contains a simple script that checks if /usr/bin/osh exists in /etc/shells, and if not it adds it there)

I'd be happy to submit a PR and maintain it, if I can get it to build.
I edited the template according to your suggestions:
```shell
# Template file for 'oilshell'
pkgname=oilshell
version=0.6.0
revision=1
wrksrc=oil-${version}
#archs="i686 x86_64"
build_style=configure
configure_args="--prefix=/usr --with-readline"
makedepends="readline-devel"
short_desc="Oil, a new unix shell."
maintainer=" <>"
license="Apache-2.0"
homepage="http://www.oilshell.org/"
distfiles="https://${pkgname}.org/download/oil-${version}.tar.xz"
checksum=141e3a5c1997a696d65747966978aaa38921d77e303aad9e77b4ab4aedab84b7

do_install() {
	vbin _bin/oil.ovm
 	# symlink osh, oshc to oil.ovm.
	cd ${DESTDIR}/usr/bin
	ln -s oil.ovm osh
	ln -s oil.ovm oshc 
	# for some reason osh expects itself to be in /usr/local/bin
	mkdir -p "${DESTDIR}/usr/local/bin"
	ln -s "${DESTDIR}/usr/bin/oil.ovm" "${DESTDIR}/usr/local/bin/osh"
}
```
I'm not sure if oilshell supports i686 or not. If the project doesn't mention anything about support, is it safe to assume there is no i686 support, or should I ask? For `configure_args`, by default it uses `--prefix=/usr`. Does adding a flag to that (`--with-readline`) overwrite the existing args? (in other words, do I need to add `--prefix=/usr`?)

It's failing to build with the message
```
=> oilshell-0.6.0_1: running do_build ...
strip -o _build/oil/ovm-opt.stripped _build/oil/ovm-opt  # What's the difference with debug symbols?
strip-wrapper: ignoring arguments: -o _build/oil/ovm-opt.stripped _build/oil/ovm-opt
# We need a relative path since it will be _bin/oil.ovm
objcopy --add-gnu-debuglink=_build/oil/ovm-opt.symbols _build/oil/ovm-opt.stripped
objcopy: '_build/oil/ovm-opt.stripped': No such file
make: *** [Makefile:157: _build/oil/ovm-opt.stripped] Error 1
=> ERROR: oilshell-0.6.0_1: do_build: '${make_cmd} ${makejobs} ${make_build_args} ${make_build_target}' exited with 2
=> ERROR:   in do_build() at common/build-style/configure.sh:14
```
Oilshell doesn't support out-of-tree builds, could that be causing this failure?

I added 
```
pre_build() {
	rm -f ${XBPS_WRAPPERDIR}/strip
}
```
like found in the grub template and it gave me an error about /usr/local being forbidden like you mentioned, so I deleted that chunk and re-ran the build and it works. I installed the package with `sudo xbps-install --repository=hostdir/binpkgs oilshell` and that works too. The issue that caused me to symlink to /usr/local in the first place isn't occurring.

I tried to compile it to i686 with `./xbps-src -a i686 pkg oilshell` but readline-devel never gets installed, so it fails. If I remove `--with-readline` from the configure-args if fails with lots of
```
#error "LONG_BIT definition appears wrong for platform (bad gcc/glibc config?)."
```
and thenobjcopy --only-keep-debug _build/oil/ovm-opt _build/oil/ovm-opt.symbols
objcopy: '_build/oil/ovm-opt': No such file
make: *** [Makefile:150: _build/oil/ovm-opt.symbols] Error 1
=> ERROR: oilshell-0.6.0_1: do_build: '${make_cmd} ${makejobs} ${make_build_args} ${make_build_target}' exited with 2
=> ERROR:   in do_build() at common/build-style/configure.sh:14
```
So I don't think it supports i686. I attached the whole log regardless
```

If I add /usr/bin/osh to /etc/shells and chsh to it, I can't log into tty2, though all other ttys work. I can log into it with root, or if I chsh back to /bin/bash

--- a/srcpkgs/oil/template
+++ b/srcpkgs/oil/template
@@ -1,0 +1,37 @@
+# Template file for 'oil'
+pkgname=oil
+version=0.6.0
+revision=1
+build_style=configure
+configure_args="--prefix=/usr $(vopt_with readline)"
+makedepends="$(vopt_if readline readline-devel)"
+short_desc="Oil, a new unix shell"
+maintainer="Caleb JA <skywise@tutanota.de>"
+license="Apache-2.0"
+homepage="https://www.oilshell.org"
+distfiles="${homepage}/download/${pkgname}-${version}.tar.xz"
+checksum=141e3a5c1997a696d65747966978aaa38921d77e303aad9e77b4ab4aedab84b7
+register_shell="/usr/bin/osh"
+nocross="oil is nocross"
+nostrip=yes
+
+build_options="readline"
+build_options_default="readline"
+
+pre_build() {
+	# Remove the strip wrapper
+	rm -f "${XBPS_WRAPPERDIR}"/strip
+}
+
+do_install() {
+	# Install binaries
+	vbin _bin/oil.ovm
+	# Symlink osh, oshc to oil.ovm.
+	cd "${DESTDIR}"/usr/bin
+	ln -s oil.ovm osh
+	ln -s oil.ovm oshc
+}
+
+post_install() {
+	vlicense LICENSE.txt
+}

